packages/theme/src/launcher.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2018-2020 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Launcher, ILauncher } from '@jupyterlab/launcher';
+
+import { each } from '@lumino/algorithm';
+
+import * as React from 'react';
+
+/**
+ * The known categories of launcher items and their default ordering.
+ */
+const KNOWN_CATEGORIES = ['Notebook', 'Console', 'Elyra', 'Other'];
+
+export class CustomLauncher extends Launcher {
+  /**
+   * Construct a new launcher widget.
+   */
+  constructor(options: ILauncher.IOptions) {
+    super(options);
+  }
+
+  /**
+   * Render the launcher to virtual DOM nodes.
+   */
+  protected render(): React.ReactElement<any> | null {
+    // Bail if there is no model.
+    if (!this.model) {
+      return null;
+    }
+
+    // get the rendering from JupyterLab Launcher
+    // and resort the categories
+    const launcherBody = super.render();
+    const launcherContent = launcherBody.props.children;
+    const launcherCategories = launcherContent.props.children;
+
+    const categories: React.ReactElement<any>[] = [];
+
+    // Assemble the final ordered list of categories
+    // based on KNOWN_CATEGORIES.
+    each(KNOWN_CATEGORIES, (category, index) => {
+      React.Children.forEach(launcherCategories, cat => {
+        if (cat.key === category) {
+          categories.push(cat);
+        }
+        if (cat.key === 'Elyra') {
+          // change the icon here...
```

Comment:
the category icon is taken from the [first item in a category](https://github.com/jupyterlab/jupyterlab/blob/master/packages/launcher/src/index.tsx#L201,L210) hence the default to the pipeline editor icon. 

one possible option to change the icon is to use the `iconClass` (property of the command) and set the image accordingly via CSS rather than try to search & replace via the children props

Reply:
I am ok with that, are you volunteering to help on the css side?

Reply:
i can take a look

Reply:
i have pushed the icon update (https://github.com/elyra-ai/elyra/pull/782/commits/35f26a2c379485b5a1b2220b876df5c25f63957a) into this PR:

---
<img width="400" alt="image" src="https://user-images.githubusercontent.com/13156555/88686716-5c116500-d0c5-11ea-9f56-6cfc2896cbe7.png">

---
<img width="400" alt="image" src="https://user-images.githubusercontent.com/13156555/88686794-73505280-d0c5-11ea-8665-59ebde3d471f.png">